content/blog/2022-07-05-Practical-data-drift-monitoring-with-DVC.md

```diff
@@ -0,0 +1,337 @@
+Alibi-detect library categorizes it as an outlier detector. It works with
+time-series sequences of data. First, it learns a pattern on a batch of training
+data. Then, during the inference phase, it classifies whether the time-series
+sequence is an outlier or whether it recognizes its pattern. In our case, one
+time-series sequence represents one day of data. We will classify each day as an
+outlier/non-outlier, and if we see multiple subsequent outliers, it may signal
+that the underlying pattern has changed. This way, we will use an outlier
+detector to check for concept drift.
+
+In fact, one of the main difficulties in the drift/outlier detection is that
+usually, we can’t use the detection algorithms out of the box as in our case.
+But we need to understand what is in the data and apply the algorithm
+accordingly.
+
+## DVC pipeline
+
+To use the detectors, we need to train them first. The training process is very
+similar to the training of the ML model. Typically, this process consists of 1)
+preparing training/test datasets, 2) running the training, and 3) saving the
+trained detector for later use.
+
+Although these steps can be done ‘manually’, for example, in a Jupyter notebook,
+it quickly becomes challenging to manage the whole process. Particularly if we
+would like to use the same training dataset for the ML model and the detectors.
+Fortunately, we have a tool called DVC that is a great helper for these
+scenarios as we can utilize its three core features. That is:
+
+### 1) Create an ML system pipeline
+
+DVC allows us to create an ML training pipeline that we can run from the
+beginning to the end with a single command. We can implement the drift detector
+training as part of the pipeline. Then, every time we train a new ML model, we
+also train a new drift detector for the input data. This saves us from having
+multiple scripts that would have to be run manually.
+
+### 2) Using dvc exp for comparison of different drift detectors
+
+The `dvc exp` command helps us easily compare the performance of different drift
+detector configurations during the training phase.
+
+### 3) Versioning and collaboration on drift detectors
+
+In some cases, we may want to train several different drift detectors and use
+them either interchangeably or use multiple of them in the production. It may
+even happen that another team member prepares each drift detector. DVC helps us
+in this effort by allowing us to easily version the detectors and share them
+with our colleagues.
+
+The ML pipeline we will create with DVC will consist of several stages. In our
+case, we will be utilizing the input dataset for both preparing data for an ML
+model as well as drift detectors. Therefore, it makes sense to keep all the data
+preprocessing in a single stage. Training of ML model and drift detectors,
+however, are independent actions and we may put them into dedicated stages. The
+overall pipeline is shown in the figure below
+
+```bash
```

Comment:
@iterative/websites  is it `cli` or `bash` now?

Reply:
`cli` as it highlights our tools command also including `bash`.